src/util/amount.rs

```diff
@@ -0,0 +1,789 @@
+
+        if negative {
+            value *= -1;
+        }
+
+        Ok(Amount::from_sat(value))
+    }
+
+    /// Express this Amount as a floating-point value in the given denomination.
+    ///
+    /// Please be aware of the risk of using floating-point numbers.
+    pub fn as_float_denom(&self, denom: Denomination) -> f64 {
+        (self.as_sat() as f64) * 10_f64.powi(denom.precision())
+    }
+
+    /// Convert an amount in floating-point notation with a given denomination.
+    ///
+    /// Please be aware of the risk of using floating-point numbers.
+    pub fn from_float_denom(value: f64, denom: Denomination) -> Amount {
+        let amt = value * 10_f64.powi(-denom.precision());
+        Amount::from_sat(if value < 0.0 {
+            (amt - 0.5) as i64
+        } else {
+            (amt + 0.5) as i64
+        })
+    }
+
+    /// Format the value of this Amount in the given denomination.
+    ///
+    /// Does not include the denomination.
+    pub fn fmt_value_in(&self, f: &mut fmt::Write, denom: Denomination) -> fmt::Result {
+        if denom.precision() > 0 {
+            // add decimal point and zeroes
+            let width = denom.precision() as usize;
+            write!(f, "{}.{:0width$}", self.as_sat(), 0, width = width)?;
+        } else if denom.precision() < 0 {
+            // need to inject a comma in the numbered
+            let nb_decimals = denom.precision().abs() as usize;
+            let real = format!("{:0width$}", self.as_sat(), width = nb_decimals);
+            if real.len() == nb_decimals {
+                write!(f, "0.{}", &real[real.len() - nb_decimals..])?;
+            } else {
+                write!(
+                    f,
+                    "{}.{}",
+                    &real[0..(real.len() - nb_decimals)],
+                    &real[real.len() - nb_decimals..]
+                )?;
+            }
+        } else {
+            // denom.precision() == 0
+            write!(f, "{}", self.as_sat())?;
+        }
+        Ok(())
+    }
+
+    /// Get a formatted string of this Amount in the given denomination.
+    ///
+    /// Does not include the denomination.
+    pub fn to_string_in(&self, denom: Denomination) -> String {
+        let mut buf = String::new();
+        self.fmt_value_in(&mut buf, denom).unwrap();
+        buf
+    }
+
+    /// Get a formatted string of this Amount in the given denomination, suffixed with the
+    /// abbreviation for the denomination.
+    pub fn to_string_denom(&self, denom: Denomination) -> String {
+        let mut buf = String::new();
+        self.fmt_value_in(&mut buf, denom).unwrap();
+        write!(buf, " {}", denom).unwrap();
+        buf
+    }
+}
+
+impl fmt::Debug for Amount {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Amount({} satoshi)", self.as_sat())
+    }
+}
+
+// No one should depend on a binding contract for Display for this type.
+// Just using Bitcoin denominated string.
+impl fmt::Display for Amount {
```

Comment:
Display is certainly useful. FromStr, I can agree that it's doubtable. Because in all cases, you'll probably care about what format is allowed. (While with Display often times (logging, printing) you don't really care.)

Reply:
We can add any number of alternative spellings to `FromStr for Denomination`. But I know what you mean. It's not exact science. It's mostly useful to make sure you can at least roundtrip your own strings. Which I think can have some value.

Reply:
I added `sat` parsing.

Reply:
Well, to be honest, `Display` and `Debug` are meant for human readability. So all you should expect from them is that you'll be able to understand the amount from seeing it's string representation.
I don't think just a number would be sufficient for `Display`. There is `as_sat()` is you just want a satoshi number and `to_string_in` if you want just the number in any other denomination. If you want I could add a shorthand `as_btc()` that returns `as_float_denom(Denomination::Bitcoin)`, but I think most people wanting the bitcoin value is just for printing, so `to_string[_in]()` would do.